src/AggregateFunctions/AggregateFunctionGroupBitmapData.h

```diff
@@ -579,6 +579,59 @@ class RoaringBitmapWithSmallSet : private boost::noncopyable
         }
     }
 
+    UInt64 rb_offset_limit(UInt64 offset, UInt64 limit, RoaringBitmapWithSmallSet & r1) const
+    {
+        if (limit == 0 || offset >= size())
+            return 0;
+
+        if (isSmall())
+        {
+            UInt64 offset_count = 0;
+            std::vector<T> answer;
+            for (const auto & x : small)
+            {
+                T val = x.getValue();
+                if (offset_count >= offset)
+                {
+                    answer.push_back(val);
+                }
+                else
+                {
+                    offset_count++;
+                }
+            }
+            if (limit < answer.size())
+            {
+                std::nth_element(answer.begin(), answer.begin() + limit, answer.end());
+                answer.resize(limit);
+            }
+
+            for (const auto & elem : answer)
+                r1.add(elem);
+            return answer.size();
+        }
+        else
+        {
+            UInt64 count = 0;
+            UInt64 offset_count = 0;
+            for (auto it = rb->begin(); it != rb->end(); ++it)
```

Comment:
Maybe better to split into to small loops?
```suggestion
            auto it = rb->begin();
            for (;it != rb->end() && offset_count < offset; ++it)
                ++offset_count;

            for (;it != rb->end() && count < limit; ++it, ++count)
                    r1.add(*it);
```

In `rb_limit` it was needed because of finding all values bigger than start value. You can just simply move iterator on offset steps forward.